synapse/api/errors.py

```diff
@@ -226,11 +232,11 @@ class InteractiveAuthIncompleteError(Exception):
     (This indicates we should return a 401 with 'result' as the body)
 
     Attributes:
-        result (dict): the server response to the request, which should be
+        result: the server response to the request, which should be
             passed back to the client
     """
 
-    def __init__(self, result):
+    def __init__(self, result: Dict):
```

Comment:
```suggestion
    def __init__(self, result: JsonDict):
```

or just `dict`

Reply:
Ugh, this makes a cycle. Good call though! 👍